apps/automation-v2/src/tlc/batch-liquidate.ts

```diff
@@ -0,0 +1,217 @@
+      gasLimit: 1_000_000n,
+    });
+    await checkAccPosition(accBatch); // check accs position after liquidation
+
+    const txReceipt = await tx.wait();
+    if (!txReceipt) return taskSuccessSilent();
+
+    // Grab the events
+    const events: TempleTaskDiscordEvent[] = [];
+
+    await txReceipt.logs.forEach((log) => {
+      if (log instanceof EventLog) {
+        // TODO: delete following interesRateUpdateEv logs once bot is working fine on sepolia
+        ctx.logger.info(`event log fragment name: ${log.fragment.name}`);
+        // test InterestRateUpdate filter is working
+        const interesRateUpdateEv = matchAndDecodeEvent(
+          log,
+          tlc,
+          tlc.filters.InterestRateUpdate()
+        );
+        if (interesRateUpdateEv) {
+          ctx.logger.info(
+            `newInterestRate: ${interesRateUpdateEv.newInterestRate}`
+          );
+        }
+
+        const liquidatedEv = matchAndDecodeEvent(
+          log,
+          tlc,
+          tlc.filters.Liquidated()
+        );
+
+        // check if this is the liquidated event
+        if (liquidatedEv) {
+          events.push({
+            what: 'Liquidated',
+            details: [
+              `account = \`${liquidatedEv.account}\``,
+              `collateralValue = \`${formatBigNumber(
+                liquidatedEv.collateralValue,
+                18,
+                4
+              )}\``,
+              `collateralSeized = \`${formatBigNumber(
+                liquidatedEv.collateralSeized,
+                18,
+                4
+              )}\``,
+              `daiDebtWiped = \`${formatBigNumber(
+                liquidatedEv.daiDebtWiped,
+                18,
+                4
+              )}\``,
+            ],
+          });
+        }
+      }
+    });
+
+    // if no liquidation happened, success silently
+    if (events.length === 0) return taskSuccessSilent();
+
+    const txUrl = config.CHAIN.transactionUrl(txReceipt.hash);
+    const metadata: TempleTaskDiscordMetadata = {
+      title: 'TLC Batch Liquidate',
+      events,
+      submittedAt,
+      txReceipt,
+      txUrl,
+    };
+
+    // Send notification
+    const message = await buildTempleTasksDiscordMessage(
+      provider,
+      config.CHAIN,
+      metadata
+    );
+    await discord.postMessage(message);
+
+    // Send discord alert if signer wallet doesn't have sufficient eth balance
+    const ethBalance = await provider.getBalance(walletAddress);
+    if (ethBalance < config.MIN_ETH_BALANCE) {
+      const ethBalanceMessage = await buildDiscordMessageCheckEth(
+        config.CHAIN,
+        submittedAt,
+        walletAddress,
+        ethBalance,
+        config.MIN_ETH_BALANCE
+      );
+
+      await discord.postMessage(ethBalanceMessage);
+    }
+  });
+
+  return taskSuccess();
+}
+
+const getTlcUsers = async (url: string) => {
+  const resp = await subgraphRequest<GetUserResponse>(url, {
+    query: `{
+      users {
```

Comment:
Could potentially paginate in the subgraph request too, eg

```
{
  users(first: 10, skip: 10) {
    ... on TlcUser {
      id
...
```

but given it's expected to be a low number of users, it's probably ok

if doing this, should also get the `block` on the first page, so then you use that same block for the subsequent pages to ensure consistency